feat: Add list_targets and add_file_to_targets to actions

srydell opened the pull request:
Hi!

First of all, thank you for this amazing plugin. I'm having a lot of fun with it :)

I wanted to add files/directories non interactively via autocmds when creating new swift files so I added some small helper functions. Hopefully they are general enough.

However this is as far as I got, I couldn't get `list_targets` to return anything, even though it's identical to the call in `run_select_targets`. Perhaps you see something I don't? And if you like the changes we can merge them.

wojciech-kulik commented:
> I wanted to add files/directories non interactively via autocmds when creating new swift files so I added some small helper functions. Hopefully they are general enough.

I refactored the code. Could you explain what you are trying to achieve with it? There is already some integration with nvim-tree to add files automatically :).

> First of all, thank you for this amazing plugin. I'm having a lot of fun with it :)

Great to hear, thank you! 🍻

srydell commented on lua/xcodebuild/actions.lua:
```diff
@@ -257,6 +261,10 @@ function M.delete_current_file()
 
+function M.list_targets()
+  projectManager.list_targets()
+end
```
Haha amazing what happens after a long day of coding : )

srydell commented:
> I refactored the code and fixed some bugs, I think in this form it could be merged.

Cool!

> Could you explain what you are trying to achieve with it? There is already some integration with nvim-tree to add files automatically :). Are you working on some automation to generate some skeleton with multiple files?

Well done with the nvim-tree integration. I'm however not currently using nvim-tree, and unfortunately it doesn't match how I work.

You are absolutely right. I've ported my old vim-skeleton plugin from using Ultisnips to luasnips and am working to create some boilerplate generation for swift. Currently the automatic adding looks like this:

```lua
local function is_xcode_project(filepath)
  local lsp_util = require('lspconfig.util')
  return lsp_util.root_pattern('buildServer.json')(filepath)
    or lsp_util.root_pattern('*.xcodeproj', '*.xcworkspace')(filepath)
    or lsp_util.root_pattern('Package.swift')(filepath)
end

-- Automatically adds the file and group to XCode
vim.api.nvim_create_autocmd({ 'BufNewFile' }, {
  pattern = '*.swift',
  group = "srydell_swift",
  callback = function()
    local actions = require('xcodebuild.actions')
    local filepath = vim.fn.expand('%:p')
    if is_xcode_project(filepath) then
      actions.add_file_to_targets(filepath, actions.list_targets())
    end
  end,
})
```

What do you think? :)

wojciech-kulik commented:
Looks good! Ok, then I’m merging these changes. Make sure to adjust your calls, because I changed some names. Thank you for the first contribution 🍻
